### PR TITLE
Fixed and changed how we seed database tables.

### PR DESCRIPTION
**Pull request by CharliedeanCS**
Removed the old seed-tables scripts so that know the entire database is added to through the load script.

The new load script goes through each line and adds plants,locations and gardeners to a table if they are already not found in the table.

Changed the pipeline script to fit this change.

### CODE REVIEW

File: pipeline/load.py [hunk 1]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
     return conn
 
 
+def insert_into_location_table(connection: Connection, plant_data: pd.DataFrame) -> None:
+    """Inserts data from a pandas data frame into a SQL Server into the location table"""
```
Comment (shivani-63) [inline pipeline/load.py]:
```suggestion
    """Inserts data from a pandas data frame into a SQL Server into the location table."""
```

File: pipeline/load.py [hunk 2]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
...
+    plant_list = plant_data.values.tolist()
+
+    for plant in plant_list:
+
+        query = sql.text(
+            "SELECT location_id FROM s_delta.location WHERE region = (:region)")
+        args = ({"region": plant[9]})
+        id = connection.execute(query, args).fetchone()
+
+        if id is None:
```
Comment (shivani-63) [inline pipeline/load.py:37]: Nice catch, smart way to upload it to the table without needing to seed!

File: pipeline/load.py [hunk 3]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
     return conn
 
 
+def insert_into_location_table(connection: Connection, plant_data: pd.DataFrame) -> None:
+    """Inserts data from a pandas data frame into a SQL Server into the location table."""
+
+    plant_list = plant_data.values.tolist()
+
+    for plant in plant_list:
+
```
Comment (zandersnow14) [inline pipeline/load.py:31]: Quick question (I'm not sure if you need to) - but does there need to be a "USE plants;" sql string somewhere here?

File: pipeline/load.py [hunk 4]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
...
+                query, {"name": name, "email": email, "telephone": telephone})
+
+
+def insert_into_plant_table(conn: Connection, plant_data: list[dict]) -> None:
+    """Seed the plant table with the plant data list and relevant botanist and location ids."""
+
+    plant_data = plant_data.to_dict('records')
+
+    for plant in plant_data:
+
```
Comment (zandersnow14) [inline pipeline/load.py:82]: Again - does there need to be a "USE plants;" sql line here?

File: pipeline/load.py [hunk 5]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
...
+            query = sql.text(
+                """INSERT INTO s_delta.location (region,country,continent)
+                VALUES (:region,:continent,:country)""")
+            connection.execute(
+                query, {"region": region, "continent": continent, "country": country})
+
+
+def insert_into_botanist_table(connection: Connection, plant_data: pd.DataFrame) -> None:
+    """Inserts data from a pandas data frame into a SQL Server into the location table"""
```
Comment (zandersnow14) [inline pipeline/load.py]:
```suggestion
    """Inserts data from a pandas data frame into a SQL Server into the location table."""
```

File: pipeline/load.py [hunk 6]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
...
+
+    plant_list = plant_data.values.tolist()
+
+    for plant in plant_list:
+
+        connection.execute(sql.text("USE plants;"))
+
+        query = sql.text(
+            "SELECT botanist_id FROM s_delta.botanist WHERE telephone_number = (:telephone)")
+        args = ({"telephone": plant[8]})
```
Comment (shivani-63) [inline pipeline/load.py]: Just a reallyyy nitpicky thing, in the SQL schema, I think Zander set the email address as the unique constraint, but here we're using the phone number. I know they're both unique, but maybe worth just going with one for standardisation :)

File: pipeline/load.py [hunk 7]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
...
+            query = sql.text(
+                "INSERT INTO s_delta.plant (plant_id, name, botanist_id, location_id) VALUES (:id, :name, :b_id, :l_id)")
+            args = ({"id": plant["Id"], "name": plant["Name"], "b_id": botanist_id,
+                    "l_id": location_id})
+            conn.execute(query, args)
+
+
 def insert_into_recordings_table(connection: Connection, plant_data: pd.DataFrame) -> None:
-    """Inserts data from a pandas data frame into a SQL Server"""
+    """Inserts data from a pandas data frame into a SQL Server into the recording table"""
```
Comment (shivani-63) [inline pipeline/load.py]:
```suggestion
    """Inserts data from a pandas data frame into a SQL Server into the recording table."""
```

File: pipeline/load.py [hunk 8]
```diff
@@ -21,14 +22,100 @@ def create_database_connection(config: _Environ) -> Connection:
...
+
+        query = sql.text(
+            "SELECT plant_id FROM s_delta.plant WHERE name = (:name)")
+        args = ({"name": plant["Name"]})
+        id = conn.execute(query, args).fetchone()
+
+        if id is None:
+
+            query = sql.text(
+                "SELECT botanist_id FROM s_delta.botanist WHERE telephone_number = (:num)")
```
Comment (zandersnow14) [inline pipeline/load.py]: Like Shivani said earlier, maybe change this to email just so everything's standardised!

File: pipeline/load.py [hunk 9]
```diff
@@ -5,6 +5,7 @@
 
 from dotenv import load_dotenv
 from sqlalchemy import create_engine, sql, Connection
+from sqlalchemy.exc import IntegrityError, ProgrammingError
```
Comment (shivani-63) [inline pipeline/load.py:8]: Really good error imports! Nice and specific :)

File: pipeline/load.py (hunk 3 above)
Comment (CharliedeanCS) [reply pipeline/load.py:31]: Only the first function being ran requires the use plants. From that point the connection which is the same throughout will be using plants

**Review by shivani-63 — APPROVED** [review]
Looks great, thank you!

**Pull request merged**